### PR TITLE
Improved WiiVC check to avoid false positives & added functions to read and manipulate fcr31

### PR DESCRIPTION
**Pull request by mpharoah**
Added functions to manage the current floating point rounding mode and exceptions. Also improved the WiiVC check so that hacks using alternate rounding modes don't get incorrectly detected as WiiVC.

Tested on console (as in N64 console) and all relevant PC emulators, but I'll need someone to test on an actual Wii to verify that the check still works as expected.

### CODE REVIEW

**Review by gheskett — APPROVED** [review]

File: asm/fcr31.s [hunk 1]
```diff
@@ -0,0 +1,69 @@
+.include "macros.inc"
+.section .text, "ax"
+.set noreorder
+.set noat
+
+glabel fcr_get_rounding_mode
+cfc1 $v0, $31
+jr $ra
+andi $v0, 0x3
```
Comment (someone2639) [inline asm/fcr31.s:9]: Only comment I have here is that these masks and flags have pretty names in PR/R4300.h with _LANGUAGE_ASSEMBLY defined (e.g. this 0x3 is `FPCSR_RM_MASK`)

Comment (someone2639) [reply asm/fcr31.s:9]: actually this might be the only one thats easy to drop in lol

**Review by someone2639 — APPROVED** [review]
no reason to request changes over one sdk name so yeah

**Review by gheskett — CHANGES_REQUESTED** [review]
Could we move `float.h` into the include folder or somewhere that's not the game folder?

**Comment by mpharoah** [comment]
Done

**Review by thecozies — APPROVED** [review]

**Review by gheskett — APPROVED** [review]

**Pull request merged**